tests/Feature/ImportResultsTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Models\Mismatch;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Tests\TestCase;
+use App\Services\ImportResults;
+use Illuminate\Support\Facades\Storage;
+use App\Models\User;
+use App\Models\ImportMeta;
+
+class ImportResultsTest extends TestCase
+{
+    use RefreshDatabase, WithFaker;
+    /**
+     * @return void
+     */
+    public function test_generate_csv_writes_to_path(): void
+    {
+        $user = User::factory()->uploader()->create();
+        $import = ImportMeta::factory()->for($user)->create();
+
+        $mismatches = Mismatch::factory(3)
+            ->for($import)
+            ->create();
+
+        Storage::fake('local');
+        $filename = 'temp_test.csv';
+
+        Storage::put($filename, '');
+        $path = Storage::path($filename);
+
+        $this->travelTo(now()); // stop the clock
```

Comment:
Do we need the time travel? As far as I can tell, nothing in the CSV depends on it, and the test still seems to pass without it.

Reply:
removed